LFS171x/docs/introduction-to-hyperledger-composer.md

```diff
@@ -0,0 +1,976 @@
+
+```
+'use strict';
+/**
+ * Defining the namespace for the business network
+ */
+const NS = 'org.tuna';
+```
+
+The transaction logic is defined in a function that accepts the Transaction `SellTuna` as input parameter.
+```
+/**
+* Transfer tuna from one owner to another
+* @param {org.tuna.SellTuna} tx - The transferTuna transaction
+* @transaction
+*/
+async function sellTuna(tx) {
+```
+
+Next, the registries related to the Asset `Tuna` and the Participant `Restaurant Owner` are instantiated:
+```
+    // Get asset registry for Tuna
+    const tunaRegistry = await getAssetRegistry(NS + '.Tuna');
+
+    // Get participant registry for Individuals
+    const restaurantOwnerRegistry = await getParticipantRegistry(NS + '.RestaurantOwner');
+```
+
+Next, we have to verify that the `status` of the Tuna is `CAUGHT`.
+This is to make sure that a `Tuna` already sold cannot be sold again.
+```
+    // Make sure the tuna status is CAUGHT and not PURCHASED
+    if (tx.tuna.status !== 'CAUGHT') {
+        throw new Error(`Tuna with id ${tx.tuna.getIdentifier()} is not in CAUGHT status`);
+    }
+```
+
+Retrieve the id of the `RestaurantOwner` from the Transaction.
+```
+    // Get restaurantOwner ID
+    const restaurantOwnerId = tx.restaurantOwner.getIdentifier();
+```
+
+Next, we verify that the `RestaurantOwner` exists
+```
+    // Make sure that restaurantOwner exists
+    const restaurantOwner = await restaurantOwnerRegistry.get(restaurantOwnerId);
+    if (!restaurantOwner) {
+        throw new Error(`RestaurantOwner with id ${restaurantOwnerId} does not exist`);
+    }
+```
+
+Now we can update the `owner` of the Tuna:
+```
+    // Update tuna with new owner
+    tx.tuna.owner = tx.restaurantOwner;
+```
+
+Update the `status` of the Tuna:
+```
+    tx.tuna.status = 'PURCHASED';
+```
+
+Update the record of the Tuna in the asset registry:
+```
+    // Update the asset in the asset registry.
+    await tunaRegistry.update(tx.tuna);
+}
+```
+
+Create the Event `TunaSale`:
+```
+    // Create a Tuna Sale Event
+    let tunaSaleEvent = getFactory().newEvent(NS, 'TunaSale');
+    tunaSaleEvent.tunaId = tx.tuna.tunaId;
+    tunaSaleEvent.restaurantName = tx.restaurantOwner.restaurantName;
+```
+
+Finally, emit the event created.
+```
+    // Emit the Event
+    emit(tunaSaleEvent);
+```
+
+## 5) Developing Queries
+The queries can be specified under the `queries.qry` file.
+
+The query `getTunaByParticipant` will return all the fishes owned by a specific participant in the format of an array of Assets of type `Tuna`.
+```
+query getTunaByParticipant {
+    description: "List tuna owned by specific 'owner'"
+    statement:
+        SELECT org.tuna.Tuna
+            WHERE (owner == _$owner)
+                ORDER BY [catchTime ASC]
+}
+```
+
+## 6) Defining Access Control Rules
+
+The Access Control Rules are defined in the file `permissions.acl`.
+
+The following rule allows only the owner of the tuna to execute the transaction `SellTuna`:
```

Comment:
Add this to the already existing rules in the `permissions.acl`.

Reply:
The way this is written now, people may think they should replace all the entries that are currently there. I don't think that is what you want.

Reply:
Indeed, fixed, by explicitly specifying we add the rule.